pymilvus/bulk_writer/remote_bulk_writer.py

```diff
@@ -0,0 +1,155 @@
+            http_client: Any = None,
+            credentials: Any = None,
+        ):
+            self._bucket_name = bucket_name
+            self._endpoint = endpoint
+            self._access_key = access_key
+            self._secret_key = secret_key
+            self._secure = (secure,)
+            self._session_token = (session_token,)
+            self._region = (region,)
+            self._http_client = (http_client,)  # urllib3.poolmanager.PoolManager
+            self._credentials = (credentials,)  # minio.credentials.Provider
+
+    def __init__(
+        self,
+        schema: CollectionSchema,
+        remote_path: str,
+        local_path: str,
+        connect_param: ConnectParam,
+        segment_size: int = 512 * MB,
+    ):
+        super().__init__(schema, local_path, segment_size)
+        uid = Path(super().data_path).name
+        self._remote_path = Path("/").joinpath(remote_path).joinpath(uid)
+        self._connect_param = connect_param
+        self._client = None
+        self._get_client()
+        logger.info(f"Remote buffer writer initialized, target path: {self._remote_path}")
+
+    def _get_client(self):
+        try:
+            if self._client is None:
+
+                def arg_parse(arg: Any):
+                    return arg[0] if isinstance(arg, tuple) else arg
+
+                self._client = Minio(
+                    endpoint=arg_parse(self._connect_param._endpoint),
+                    access_key=arg_parse(self._connect_param._access_key),
+                    secret_key=arg_parse(self._connect_param._secret_key),
+                    secure=arg_parse(self._connect_param._secure),
+                    session_token=arg_parse(self._connect_param._session_token),
+                    region=arg_parse(self._connect_param._region),
+                    http_client=arg_parse(self._connect_param._http_client),
+                    credentials=arg_parse(self._connect_param._credentials),
+                )
+            else:
+                return self._client
+        except Exception as err:
+            logger.error(f"Failed to connect MinIO/S3, error: {err}")
+            raise
+
+    def append(self, row: dict, **kwargs):
+        super().append(row, **kwargs)
+
+    def commit(self, **kwargs):
+        super().commit(call_back=self._upload)
+
+    def _remote_exists(self, file: str) -> bool:
+        try:
+            minio_client = self._get_client()
+            minio_client.stat_object(bucket_name=self._connect_param._bucket_name, object_name=file)
+        except S3Error as err:
+            if err.code == "NoSuchKey":
+                return False
+            self._throw(f"Failed to stat MinIO/S3 object, error: {err}")
+        return True
+
+    def _local_rm(self, file: str):
+        try:
+            Path.unlink(file)
+        except Exception:
+            logger.warning(f"Failed to delete local file: {file}")
+
+    def _upload(self, file_list: list):
```

Comment:
It is recommended to utilize RESTful interfaces for manipulating object storage in this context.

Reply:
not in current version